--- a/src/owiwind.F
+++ b/src/owiwind.F
status=modified +1 -1
@@ -23,7 +23,7 @@
 !-----------------------------------------------------------------------
 !  MODULE OWIWIND
 !-----------------------------------------------------------------------
-!> @author Zachary Cobell, The Water Institute, zcobell@thewaterinstitute.org
+!> @author Zachary Cobell, ICEYE, zach.cobell@iceye.com
 !> @author Shintaro Bunya
 !> 
 !> @copyright Dr. R.A. Luettich and Dr. J.J. Westerink
