--- a/src/netcdfio.F90
+++ b/src/netcdfio.F90
status=modified +27 -27
@@ -675,12 +675,12 @@ subroutine initStationFile(sta, descript1, reterr)
          iret = nf90_def_var(sta%ncid, 'sigmat', NF90_DOUBLE, &
                              sta%station_data_dims_3D, sta%u_station_data_id)
          call check_err(iret)
-         if ((IDEN == 2) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
             iret = nf90_def_var(sta%ncid, 'salinity', NF90_DOUBLE, &
                                 sta%station_data_dims_3D, sta%v_station_data_id)
             call check_err(iret)
          end if
-         if ((IDEN == 3) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
             iret = nf90_def_var(sta%ncid, 'temperature', NF90_DOUBLE, &
                                 sta%station_data_dims_3D, sta%w_station_data_id)
             call check_err(iret)
@@ -699,7 +699,7 @@ subroutine initStationFile(sta, descript1, reterr)
          call putUnitsAttribute(sta%ncid, sta%u_station_data_id, &
                                 "kg/m^3", "n/a")
          ! salinity
-         if ((IDEN == 2) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
             att_text = &
                'station water column vertically varying salinity'
             iret = nf90_put_att(sta%ncid, &
@@ -715,7 +715,7 @@ subroutine initStationFile(sta, descript1, reterr)
                                    "PSU", "n/a")
          end if
          ! temperature
-         if ((IDEN == 3) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
             att_text = &
                'station water column vertically varying temperature'
             iret = nf90_put_att(sta%ncid, &
@@ -1365,12 +1365,12 @@ subroutine initStationFile(sta, descript1, reterr)
             iret = nf90_def_var_deflate(sta%ncid, sta%u_station_data_id, &
                                         1, 1, 2)
             call check_err(iret)
-            if ((IDEN == 2) .or. (IDEN == 4)) then
+            if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
                iret = nf90_def_var_deflate(sta%ncid, &
                                            sta%v_station_data_id, 1, 1, 2)
                call check_err(iret)
             end if
-            if ((IDEN == 3) .or. (IDEN == 4)) then
+            if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
                iret = nf90_def_var_deflate(sta%ncid, &
                                            sta%w_station_data_id, 1, 1, 2)
                call check_err(iret)
@@ -1547,12 +1547,12 @@ subroutine initNodalDataFile(dat, descript1, reterr)
          iret = nf90_def_var(dat%ncid, 'sigmat', NF90_DOUBLE, &
                              dat%nodal_data_dims_3D, dat%u_nodal_data_id)
          call check_err(iret)
-         if ((IDEN == 2) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
             iret = nf90_def_var(dat%ncid, 'salinity', NF90_DOUBLE, &
                                 dat%nodal_data_dims_3D, dat%v_nodal_data_id)
             call check_err(iret)
          end if
-         if ((IDEN == 3) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
             iret = nf90_def_var(dat%ncid, 'temperature', NF90_DOUBLE, &
                                 dat%nodal_data_dims_3D, dat%w_nodal_data_id)
             call check_err(iret)
@@ -1569,7 +1569,7 @@ subroutine initNodalDataFile(dat, descript1, reterr)
          call putUnitsAttribute(dat%ncid, dat%u_nodal_data_id, &
                                 "kg/m^3", "n/a")
          ! salinity
-         if ((IDEN == 2) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
             att_text = "water column vertically varying salinity"
             iret = nf90_put_att(dat%ncid, dat%v_nodal_data_id, &
                                 'long_name', trim(att_text))
@@ -1582,7 +1582,7 @@ subroutine initNodalDataFile(dat, descript1, reterr)
                                    "PSU", "PSU")
          end if
          ! temperature
-         if ((IDEN == 3) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
             att_text = "water column vertically varying temperature"
             iret = nf90_put_att(dat%ncid, dat%w_nodal_data_id, &
                                 'long_name', trim(att_text))
@@ -3610,12 +3610,12 @@ subroutine initNodalDataFile(dat, descript1, reterr)
             iret = nf90_def_var_deflate(dat%ncid, dat%u_nodal_data_id, &
                                         1, 1, 2)
             call check_err(iret)
-            if ((IDEN == 2) .or. (IDEN == 4)) then
+            if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
                iret = nf90_def_var_deflate(dat%ncid, dat%v_nodal_data_id, &
                                            1, 1, 2)
                call check_err(iret)
             end if
-            if ((IDEN == 3) .or. (IDEN == 4)) then
+            if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
                iret = nf90_def_var_deflate(dat%ncid, dat%w_nodal_data_id, &
                                            1, 1, 2)
                call check_err(iret)
@@ -4353,7 +4353,7 @@ subroutine writeStationData(sta, lun, descript1, timesec, &
          end if
          call check_err(iret)
 
-         if ((iden == 2) .or. (iden == 4)) then
+         if ((abs(iden) == 2) .or. (abs(iden) == 4)) then
             iret = nf90_inq_varid(sta%ncid, "salinity", &
                                   sta%v_station_data_id)
             call check_err(iret)
@@ -4367,7 +4367,7 @@ subroutine writeStationData(sta, lun, descript1, timesec, &
 
             call check_err(iret)
          end if
-         if ((iden == 3) .or. (iden == 4)) then
+         if ((abs(iden) == 3) .or. (abs(iden) == 4)) then
             iret = nf90_inq_varid(sta%ncid, "temperature", &
                                   sta%w_station_data_id)
             call check_err(iret)
@@ -4656,12 +4656,12 @@ subroutine writeNodalData(dat, lun, descript1, timesec, &
       case (44)
          iret = nf90_inq_varid(dat%ncid, "sigmat", dat%u_nodal_data_id)
          call check_err(iret)
-         if ((iden == 2) .or. (iden == 4)) then
+         if ((abs(iden) == 2) .or. (abs(iden) == 4)) then
             iret = nf90_inq_varid(dat%ncid, "salinity", &
                                   dat%v_nodal_data_id)
             call check_err(iret)
          end if
-         if ((iden == 3) .or. (iden == 4)) then
+         if ((abs(iden) == 3) .or. (abs(iden) == 4)) then
             iret = nf90_inq_varid(dat%ncid, "temperature", &
                                   dat%w_nodal_data_id)
             call check_err(iret)
@@ -4880,7 +4880,7 @@ subroutine writeNodalData(dat, lun, descript1, timesec, &
          end if
          call check_err(iret)
 
-         if ((iden == 2) .or. (iden == 4)) then
+         if ((abs(iden) == 2) .or. (abs(iden) == 4)) then
             if (MNPROC == 1) then
                iret = nf90_put_var(dat%ncid, dat%v_nodal_data_id, &
                                    descript2%array2D, start3D, kount3D)
@@ -4890,7 +4890,7 @@ subroutine writeNodalData(dat, lun, descript1, timesec, &
             end if
             call check_err(iret)
          end if
-         if ((iden == 3) .or. (iden == 4)) then
+         if ((abs(iden) == 3) .or. (abs(iden) == 4)) then
             if (MNPROC == 1) then
                iret = nf90_put_var(dat%ncid, dat%w_nodal_data_id, &
                                    descript3%array2D, start3D, kount3D)
@@ -6559,7 +6559,7 @@ subroutine initNetCDFHotstart3D(lun, netcdf_format)
          hs%density3D%nodal_data_dims_3D(2) = hs%myMesh%num_v_nodes_dim_id
          hs%density3D%nodal_data_dims_3D(3) = hs%myTime%timenc_dim_id
       end if
-      if (IDEN == 1) then
+      if (abs(IDEN) == 1) then
          iret = nf90_def_var(hs%ncid, 'sigt', NF90_DOUBLE, &
                              hs%density3D%nodal_data_dims_3D, hs%density3D%u_nodal_data_id)
          call check_err(iret)
@@ -6571,7 +6571,7 @@ subroutine initNetCDFHotstart3D(lun, netcdf_format)
                              '_FillValue', doubleval)
          call check_err(iret)
       end if
-      if ((IDEN == 2) .or. (IDEN == 4)) then
+      if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
          iret = nf90_def_var(hs%ncid, 'salinity', NF90_DOUBLE, &
                              hs%density3D%nodal_data_dims_3D, &
                              hs%density3D%v_nodal_data_id)
@@ -6584,12 +6584,12 @@ subroutine initNetCDFHotstart3D(lun, netcdf_format)
                              '_FillValue', doubleval)
          call check_err(iret)
       end if
-      if ((IDEN == 3) .or. (IDEN == 4)) then
+      if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
          iret = nf90_def_var(hs%ncid, 'temperature', NF90_DOUBLE, &
                              hs%density3D%nodal_data_dims_3D, &
                              hs%density3D%w_nodal_data_id)
          call check_err(iret)
-         att_text = "salinity"
+         att_text = "temperature"
          iret = nf90_put_att(hs%ncid, hs%density3D%w_nodal_data_id, &
                              'long_name', trim(att_text))
          call check_err(iret)
@@ -6692,17 +6692,17 @@ subroutine initNetCDFHotstart3D(lun, netcdf_format)
          iret = nf90_def_var_deflate(hs%ncid, hs%bsy%nodal_data_id, &
                                      1, 1, 2)
          call check_err(iret)
-         if (IDEN == 1) then
+         if (abs(IDEN) == 1) then
             iret = nf90_def_var_deflate(hs%ncid, &
                                         hs%density3D%u_nodal_data_id, 1, 1, 2)
             call check_err(iret)
          end if
-         if ((IDEN == 2) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 2) .or. (abs(IDEN) == 4)) then
             iret = nf90_def_var_deflate(hs%ncid, &
                                         hs%density3D%v_nodal_data_id, 1, 1, 2)
             call check_err(iret)
          end if
-         if ((IDEN == 3) .or. (IDEN == 4)) then
+         if ((abs(IDEN) == 3) .or. (abs(IDEN) == 4)) then
             iret = nf90_def_var_deflate(hs%ncid, &
                                         hs%density3D%w_nodal_data_id, 1, 1, 2)
             call check_err(iret)
@@ -7806,7 +7806,7 @@ subroutine writeNetCDFHotstart3DVar(lun, descript)
          call check_err(iret)
       case ("Temperature")
          iret = nf90_inq_varid(hs%ncid, "temperature", &
-                               hs%density3D%nodal_data_id)
+                               hs%density3D%w_nodal_data_id)
          call check_err(iret)
          if (MNPROC == 1) then
             iret = nf90_put_var(hs%ncid, &
@@ -9407,7 +9407,7 @@ subroutine readNetCDFHotstart3D(lun)
                                             nfen, hs%myMesh%num_nodes, hs%turbulence3D%u_nodal_data_id, &
                                             subdomain_reals=q20)
          call mapFullDomainToSubdomainNPByM(hs%ncid, &
-                                            nfen, hs%myMesh%num_nodes, hs%velocity3D%v_nodal_data_id, &
+                                            nfen, hs%myMesh%num_nodes, hs%turbulence3D%v_nodal_data_id, &
                                             subdomain_reals=l)
       end if
 
