--- a/thirdparty/swan/SwanCompUnstruc.ftn90
+++ b/thirdparty/swan/SwanCompUnstruc.ftn90
status=modified +16 -6
@@ -99,7 +99,8 @@ subroutine SwanCompUnstruc ( ac2, ac1, compda, spcsig, spcdir, xytst, cross, it
 !PUN    use SIZES, only: MNPROC
 !PUN    use MESSENGER
 !ADC    use couple2adcirc, only: MakeBoundariesReflective
-!ADC    use NodalAttributes, only: FoundSwanWaveRefrac, LoadSwanWaveRefrac, SwanWaveRefrac
+! Nicole 250327: add SwanLocalControl
+!ADC    use NodalAttributes, only: FoundSwanWaveRefrac, LoadSwanWaveRefrac, SwanWaveRefrac, SwanLocalControl
 !
     implicit none
 !
@@ -639,11 +640,19 @@ subroutine SwanCompUnstruc ( ac2, ac1, compda, spcsig, spcdir, xytst, cross, it
     ! marks vertices active and non-active
     !
     !$omp single
+!Nicole 250315: If statement to deactivate SwanLocalControl 0 nodes 
     nwetp = 0.
     do kvert = 1, nverts
-       if ( compda(kvert,JDP2) > DEPMIN ) then
-          vert(kvert)%active = .true.
-          nwetp = nwetp +1.
+!NADC  if ( compda(kvert,JDP2) > DEPMIN ) then
+!NADC     vert(kvert)%active = .true.
+!NADC     nwetp = nwetp +1.
+!ADC   if ( NINT(SwanLocalControl(kvert, 1)) ==1 ) then
+!ADC      if ( compda(kvert,JDP2) > DEPMIN ) then
+!ADC         vert(kvert)%active = .true.
+!ADC         nwetp = nwetp +1.
+!ADC      else
+!ADC         vert(kvert)%active = .false.
+!ADC      endif
        else
           vert(kvert)%active = .false.
        endif
@@ -846,8 +855,9 @@ subroutine SwanCompUnstruc ( ac2, ac1, compda, spcsig, spcdir, xytst, cross, it
 !ADC                IREFR = nint(SwanWaveRefrac(ivert))
 !ADC             endif
              !
-             if ( vert(ivert)%active ) then   ! this active vertex needs to be updated
-                !
+!NADC        if ( vert(ivert)%active ) then   ! this active vertex needs to be updated
+!250316 Nicole: fix to not update internal sources              
+!ADC         if ( vert(ivert)%active .and. NINT(SwanLocalControl(ivert, 2))==0 ) then   ! this active vertex needs to be updated
                 ! determine whether the present vertex is a test point
                 !
                 IPTST  = 0
